sw/device/lib/testing/i2c_testutils.c

```diff
@@ -0,0 +1,79 @@
+// Copyright lowRISC contributors.
+// Licensed under the Apache License, Version 2.0, see LICENSE for details.
+// SPDX-License-Identifier: Apache-2.0
+
+#include "sw/device/lib/testing/i2c_testutils.h"
+
+#include <assert.h>
+#include <stdbool.h>
+#include <stdint.h>
+
+#include "sw/device/lib/base/abs_mmio.h"
+#include "sw/device/lib/base/mmio.h"
+#include "sw/device/lib/dif/dif_i2c.h"
+#include "sw/device/lib/runtime/hart.h"
+#include "sw/device/lib/runtime/log.h"
+#include "sw/device/lib/testing/test_framework/check.h"
+
+#include "i2c_regs.h"  // Generated.
+
+static const uint8_t kI2cWrite = 0;
+static const uint8_t kI2cRead = 1;
+
+// default flags
+static const dif_i2c_fmt_flags_t default_flags = {.start = false,
+                                                  .stop = false,
+                                                  .read = false,
+                                                  .read_cont = false,
+                                                  .suppress_nak_irq = false};
+
+void i2c_testutils_wr(dif_i2c_t *i2c, uint8_t addr, uint8_t byte_count,
+                      uint8_t *data, bool skip_stop) {
+  dif_i2c_fmt_flags_t flags = default_flags;
+  uint8_t data_frame;
+
+  // need a max check for byte_count right now
```

Comment:
This comments seems lost

Reply:
what do you mean? do you mean you don't understand the comment? Or that there should be more description to it?

Reply:
yeah let me do that.